File: SuperFeature/PlatformView/.arkui-x/ios/app/MyPlatformViewFactory.m
#import <Foundation/Foundation.h>

#import "MyPlatformViewFactory.h"
#import "MyWebview.h"
#import "MyMapView.h"
#import "MyVideoView.h"

@implementation MyPlatformViewFactory {

}

- (instancetype)initWithFrame{
    return self;
}

- (NSObject<IPlatformView>*) getPlatformView:(NSString*) platformViewId {
    NSLog(@"[PlatformViewUI] getPlatfformView");
    if ([platformViewId isEqualToString:@"MapView"]) {
        NSObject<IPlatformView> * view = [[MyMapView alloc] initWithFrame];
        return view;
    } else if ([platformViewId isEqualToString:@"VideoView"]) {
        NSObject<IPlatformView> * view = [[MyVideoView alloc] initWithFrame];
        return view;
    }
    return nil;
}

- (NSObject<IPlatformView>*) getPlatformView:(NSString*) platformViewId data:(NSString*) data {
    if ([platformViewId isEqualToString:@"WebView"]) {
        NSObject<IPlatformView> * view = [[MyWebview alloc] initWithFrame:data];
        return view;
    }
    return nil;
}
@end

File: SuperFeature/PlatformView/.arkui-x/android/app/src/main/java/com/example/platformview/MyPlatformViewFactory.java
package com.example.platformview;

import android.content.Context;
import android.os.Bundle;

import ohos.ace.adapter.capability.platformview.IPlatformView;
import ohos.ace.adapter.capability.platformview.PlatformViewFactory;

/**
 * MyPlatformViewFactory.
 *
 * @since 2025-01-16
 */
public class MyPlatformViewFactory extends PlatformViewFactory {
    private Context context;
    private Bundle savedInstanceState;

    @Override
    public IPlatformView getPlatformView(String id) {
        if ("MapView".equals(id)) {
            return new MyMapView(context, savedInstanceState);
        } else if ("VideoView".equals(id)) {
            return new MyVideoView(context);
        }
        return null;
    }

    @Override
    public IPlatformView getPlatformView(String id, String data) {
        if ("WebView".equals(id)) {
            return new MyWebView(context, data);
        }
        return null;
    }

    public void setContext(Context context) {
        this.context = context;
    }

    public void setSavedInstanceState(Bundle savedInstanceState) {
        this.savedInstanceState = savedInstanceState;
    }
}
